Android/GoogleAndroidVending/docs/Expansion.md

```diff
@@ -0,0 +1,415 @@
+{
+    // process the stream of the contained file
+}
+```
+
+Files accessed this way can be stored in a compressed format as a decompression stream is
+provided. This stream can be processed using typical .NET means, such as providing this
+stream to a deserializer or some other reader.
+
+The `APKExpansionSupport.GetAPKExpansionZipFile` method returns a combined collection of all
+the items in either of the expansion files. We can then query this for a specific item using
+the `GetInputStream` method on the collection. We pass a relative path to the item in expansion
+file. Once the entry is returned, we can read the stream of the item.
+
+## Testing the Downloader
+
+Testing the download manager is very similar to testing the licensing. More information on this
+in the Testing the Licensing section of the Licensing documentation][4]. The downloader 
+performs the license check internally, and in the response, it receives the expansion files 
+URIs from Play. It then uses these to initiate the download.
+
+In order to be able to download the expansion files, we have to have uploaded them when updating
+the app. The first upload does not allow the expansion files to be added, but we can just 
+re-upload the same package twice. On the second time, we can upload the expansion files with the actual app package, remembering to increase the version number beforehand.
+
+We can start testing our app by checking its ability to read the expansion files. We can do this
+by placing and naming the files just as the downloader would. Because the downloader always places
+the files at the `[shared-storage]/Android/obb/[package-name]/` location, with the name
+`[main|patch].[expansion-version].[package-name].obb`, we just have to place our files there. By
+skipping the download process, we don't have to upload the app first.
+
+Once we are sure that the app can access and use the expansion files, we can upload the files
+with the app to Play and publish to any channel to test the download. Any channel can be used,
+including Alpha and Beta.
+
+## Important Things to Remember
+
+Testing the expansion files, from downloading to reading, requires that we have the various bits
+in place:
+
+ - The **app version** on the device must be the same as the app that is on the store.
+ - Make sure that the app is indeed **published**, and not in draft.
+ - Ensure that the Expansion files have been **associated with the app**.
+ - Make sure that the response in **"Settings"** is set to **"RESPOND_NORMALLY"** as this is
+   the only response that returns the expansion files.
+ - Make sure the app has all the required permissions, there are at least 6.
+
+[1]: https://www.nuget.org/packages/Xamarin.Google.Android.Vending.Expansion.Downloader
+[2]: https://www.nuget.org/packages/Xamarin.Google.Android.Vending.Expansion.ZipFile
+[3]: https://github.com/xamarin/XamarinComponents/tree/master/Android/GoogleAndroidVending
+[4]: Licensing.md
```

Comment:
Licensing.md#getting-things-ready

Does that work?  The should drop the user right at the anchor.